easier68k/core/opcodes/move.py

```diff
@@ -311,3 +315,461 @@ def from_str(cls, command: str, parameters: str):
+        """
+        Tests whether the given command is valid
+
+        >>> Movea.is_valid('MOVEA.W', 'D0, A1')[0]
+        True
+
+        >>> Movea.is_valid('MOVEA.L', 'D0, D1')[0]
+        False
+
+        >>> Movea.is_valid('MOVEA.B', 'D0, A1')[0]
+        False
+
+        >>> Movea.is_valid('MOVEA.W', 'D0')[0]
+        False
+
+        >>> Movea.is_valid('MOVEA.G', 'D0, D1')[0]
+        False
+
+        >>> Movea.is_valid('MOV.L', 'D0, D1')[0]
+        False
+
+        >>> Movea.is_valid('MOVEA.', 'D0, D1')[0]
+        False
+
+        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+        :param parameters: The parameters after the command (such as the source and destination of a move)
+        :return: Whether the given command is valid and a list of issues/warnings encountered
+        """
+        return opcode_util.n_param_is_valid(command, parameters, "MOVEA", 2, Movea.valid_sizes, param_invalid_modes=[[],
+                                            [mode for mode in EAMode if mode is not EAMode.ARD]])[:2]  # Select all but ARD
+
+    @classmethod
+    def disassemble_instruction(cls, data: bytearray) -> Opcode:
+        """
+        This has a non-move opcode
+        >>> Movea.disassemble_instruction(bytearray.fromhex('D280'))
+
+
+        MOVEA.L (A4),A7
+        >>> op = Movea.disassemble_instruction(bytearray.fromhex('2E55'))
+
+        >>> str(op.src)
+        'EA Mode: EAMode.ARI, Data: 5'
+
+        >>> str(op.dest)
+        'EA Mode: EAMode.ARD, Data: 7'
+
+        MOVEA.W #$DEAF,A2
+        >>> op = Movea.disassemble_instruction(bytearray.fromhex('347CDEAF'))
+
+        >>> str(op.src)
+        'EA Mode: EAMode.IMM, Data: 57007'
+
+        >>> str(op.dest)
+        'EA Mode: EAMode.ARD, Data: 2'
+
+        MOVE.L ($1000).W,($200000).L
+        >>> op = Movea.disassemble_instruction(bytearray.fromhex('2A781000'))
+
+        >>> str(op.src)
+        'EA Mode: EAMode.AWA, Data: 4096'
+
+        >>> str(op.dest)
+        'EA Mode: EAMode.ARD, Data: 5'
+
+        Parses some raw data into an instance of the opcode class
+        :param data: The data used to convert into an opcode instance
+        :return: The constructed instance or none if there was an error and
+            the amount of data in words that was used (e.g. extra for immediate
+            data) or 0 for not a match
+        """
+        assert len(data) >= 2, 'opcode size is at least 1 word'
+
+        # 'big' endian byte order
+        first_word = int.from_bytes(data[0:2], 'big')
+
+        [opcode_bin,
+         size_bin,
+         destination_register_bin,
+         destination_mode_bin,
+         source_mode_bin,
+         source_register_bin] = split_bits(first_word, [2, 2, 3, 3, 3, 3])
+
+        # check opcode
+        if opcode_bin != 0b00:
+            return None
+
+        # the binary will contain the MoveSize, convert this to an OpSize used by everything else
+        size = MoveSize(size_bin).to_op_size()
```

Comment:
I discovered a problem that can occur here where size_bin is not a size in MoveSize enum it will raise a ValueError which needs to be caught here (and in Move where I discovered it).